model/settings/service.go

```diff
@@ -202,10 +202,17 @@ func (s *SettingsService) ConfirmEmailUpdate(inst *instance.Instance, tok string
 		return fmt.Errorf("failed to save the settings changes: %w", err)
 	}
 
+	publicName, _ := settings.M["public_name"].(string)
+	// if the public name is not defined, use the instance's domain
+	if publicName == "" {
+		split := strings.Split(inst.DomainName(), ".")
+		publicName = split[0]
+	}
```

Comment:
Yes, but `PublicName` makes a request to CouchDB to load the settings document and I would prefer to avoid that as we already have this document loaded.